pkg/client/cache/fifo.go

```diff
@@ -39,6 +44,16 @@ type Queue interface {
 	HasSynced() bool
 }
 
+// Helper function for popping from Queue.
+func Pop(queue Queue) interface{} {
```

Comment:
I don't understand how anyone using this helper function will not have the problem you're trying to prevent?